packages/algob/src/builtin-tasks/deploy.ts

```diff
@@ -35,10 +35,10 @@ export function loadFilenames (directory: string, taskType?: string): string[] {
     }
   }
 
-  const jsFiles = glob.sync(path.join(directory, "*.js")).sort(cmpStr);
-  if (jsFiles.length) { return jsFiles; }
-  // if not js file, return the compiled ts files from ./build
-  return glob.sync(path.join("build", directory, "*.js")).sort(cmpStr);
+  // return .js, .ts files from /scripts directory
+  return glob.sync(path.join(directory, "*.js"))
+    .concat(glob.sync(path.join(directory, "*.ts")))
+    .sort(cmpStr);
```

Comment:
what is the difference between previous and current implementation ?
i think we added the code to support `htlc-ts-project`.

Reply:
By the previous implementation we were 'changing' the relative paths. Eg. `/scripts/file.ts` -> `/build/scripts/file.js`, which was affecting the `checkpoints`, `txWriter` as well. Since the path was changed everything was relative to `/build` (the checkpoints were being saved in `/artifacts/build/scripts/..`, which we don't want. (As we're handling loading the file from build internally). Also since the cp path was changed, `getSSC`, `.asa.get()` were returning `undefined`, which was the bug. So i decided to change the implementation. 

By the current one, we're not changing the path, instead we're loading directly ts, js files (which now returns `[/scripts/file.ts, ..`] instead of [`/build/scripts/file.js, ..`]. This ensures that the paths are correct (with cp, txwriter as well), checkpoints are also stored in `/artifacts/scripts/file.ts.cp.yaml`. We only load the script from build when we need to run it. If build is not present, it throws the error with stacktrace that `/build/scripts/file..` is not found (as we suggest in the readme). 

Please let me know if that makes sense.

Reply:
makes sense